### PR TITLE
poppler: enable .dll builds

### PR DESCRIPTION
**Pull request by darealshinji**
Tested on `i686-w64-mingw32.shared`. Also related to https://github.com/mxe/mxe/issues/1388

### CODE REVIEW

**Comment by starius** [comment]
Could you change format of the patch to our git-like format using `tools/patch-tool-mxe`?

**Comment by darealshinji** [comment]
Patch was updated.

File: src/poppler.mk [hunk 1]
```diff
@@ -26,6 +26,8 @@ define $(PKG)_BUILD
         ./configure \
         $(MXE_CONFIGURE_OPTS) \
         --disable-silent-rules \
+        $(if $(BUILD_STATIC),--disable-shared --enable-static,\
+            --enable-shared --disable-static) \
```
Comment (tonytheodore) [inline src/poppler.mk]: `$(MXE_CONFIGURE_OPTS)` sets `--disable-shared` etc. and these two lines aren't needed.

Comment (darealshinji) [inline src/poppler.mk]: I'll remove them.

File: src/poppler.mk [hunk 2]
```diff
@@ -64,3 +64,4 @@ define $(PKG)_BUILD
         '$(2).cxx' -o '$(PREFIX)/$(TARGET)/bin/test-poppler.exe' \
         `'$(TARGET)-pkg-config' poppler poppler-cpp --cflags --libs`
 endef
+
```
Comment (tonytheodore) [inline src/poppler.mk]: Can you remove this trailing space and rebase into a single commit?

**Comment by tonytheodore** [comment]
Installed libs look good:

```
$ ls -al usr/*/{bin,lib}/libpoppler*
-rwxr-xr-x  1 tonyt  staff  3352473 15 Jul 22:47 usr/i686-w64-mingw32.shared/bin/libpoppler-49.dll
-rwxr-xr-x  1 tonyt  staff   323578 15 Jul 22:47 usr/i686-w64-mingw32.shared/bin/libpoppler-cpp-0.dll
-rwxr-xr-x  1 tonyt  staff   802155 15 Jul 22:47 usr/i686-w64-mingw32.shared/bin/libpoppler-glib-8.dll
-rwxr-xr-x  1 tonyt  staff  1016184 15 Jul 22:47 usr/i686-w64-mingw32.shared/bin/libpoppler-qt4-4.dll
-rwxr-xr-x  1 tonyt  staff   118460 15 Jul 22:47 usr/i686-w64-mingw32.shared/lib/libpoppler-cpp.dll.a
-rwxr-xr-x  1 tonyt  staff     2628 15 Jul 22:47 usr/i686-w64-mingw32.shared/lib/libpoppler-cpp.la
-rwxr-xr-x  1 tonyt  staff   862634 15 Jul 22:47 usr/i686-w64-mingw32.shared/lib/libpoppler-glib.dll.a
-rwxr-xr-x  1 tonyt  staff     3215 15 Jul 22:47 usr/i686-w64-mingw32.shared/lib/libpoppler-glib.la
-rwxr-xr-x  1 tonyt  staff   843056 15 Jul 22:47 usr/i686-w64-mingw32.shared/lib/libpoppler-qt4.dll.a
-rwxr-xr-x  1 tonyt  staff     2649 15 Jul 22:47 usr/i686-w64-mingw32.shared/lib/libpoppler-qt4.la
-rwxr-xr-x  1 tonyt  staff  4858952 15 Jul 22:47 usr/i686-w64-mingw32.shared/lib/libpoppler.dll.a
-rwxr-xr-x  1 tonyt  staff     2480 15 Jul 22:47 usr/i686-w64-mingw32.shared/lib/libpoppler.la
-rw-r--r--  1 tonyt  staff   469242 15 Jul 22:47 usr/i686-w64-mingw32.static/lib/libpoppler-cpp.a
-rwxr-xr-x  1 tonyt  staff     3046 15 Jul 22:47 usr/i686-w64-mingw32.static/lib/libpoppler-cpp.la
-rw-r--r--  1 tonyt  staff   637436 15 Jul 22:47 usr/i686-w64-mingw32.static/lib/libpoppler-glib.a
-rwxr-xr-x  1 tonyt  staff     4029 15 Jul 22:47 usr/i686-w64-mingw32.static/lib/libpoppler-glib.la
-rw-r--r--  1 tonyt  staff   991498 15 Jul 22:47 usr/i686-w64-mingw32.static/lib/libpoppler-qt4.a
-rwxr-xr-x  1 tonyt  staff     3218 15 Jul 22:47 usr/i686-w64-mingw32.static/lib/libpoppler-qt4.la
-rw-r--r--  1 tonyt  staff  4414214 15 Jul 22:47 usr/i686-w64-mingw32.static/lib/libpoppler.a
-rwxr-xr-x  1 tonyt  staff     2901 15 Jul 22:47 usr/i686-w64-mingw32.static/lib/libpoppler.la
-rwxr-xr-x  1 tonyt  staff  4404998 15 Jul 22:49 usr/x86_64-w64-mingw32.shared/bin/libpoppler-49.dll
-rwxr-xr-x  1 tonyt  staff   526148 15 Jul 22:49 usr/x86_64-w64-mingw32.shared/bin/libpoppler-cpp-0.dll
-rwxr-xr-x  1 tonyt  staff   995736 15 Jul 22:49 usr/x86_64-w64-mingw32.shared/bin/libpoppler-glib-8.dll
-rwxr-xr-x  1 tonyt  staff  1409127 15 Jul 22:49 usr/x86_64-w64-mingw32.shared/bin/libpoppler-qt4-4.dll
-rwxr-xr-x  1 tonyt  staff   116252 15 Jul 22:49 usr/x86_64-w64-mingw32.shared/lib/libpoppler-cpp.dll.a
-rwxr-xr-x  1 tonyt  staff     2676 15 Jul 22:49 usr/x86_64-w64-mingw32.shared/lib/libpoppler-cpp.la
-rwxr-xr-x  1 tonyt  staff   843930 15 Jul 22:49 usr/x86_64-w64-mingw32.shared/lib/libpoppler-glib.dll.a
-rwxr-xr-x  1 tonyt  staff     3279 15 Jul 22:49 usr/x86_64-w64-mingw32.shared/lib/libpoppler-glib.la
-rwxr-xr-x  1 tonyt  staff   828418 15 Jul 22:49 usr/x86_64-w64-mingw32.shared/lib/libpoppler-qt4.dll.a
-rwxr-xr-x  1 tonyt  staff     2697 15 Jul 22:49 usr/x86_64-w64-mingw32.shared/lib/libpoppler-qt4.la
-rwxr-xr-x  1 tonyt  staff  4767456 15 Jul 22:49 usr/x86_64-w64-mingw32.shared/lib/libpoppler.dll.a
-rwxr-xr-x  1 tonyt  staff     2524 15 Jul 22:49 usr/x86_64-w64-mingw32.shared/lib/libpoppler.la
-rw-r--r--  1 tonyt  staff   961166 15 Jul 22:48 usr/x86_64-w64-mingw32.static/lib/libpoppler-cpp.a
-rwxr-xr-x  1 tonyt  staff     3106 15 Jul 22:48 usr/x86_64-w64-mingw32.static/lib/libpoppler-cpp.la
-rw-r--r--  1 tonyt  staff   931598 15 Jul 22:48 usr/x86_64-w64-mingw32.static/lib/libpoppler-glib.a
-rwxr-xr-x  1 tonyt  staff     4117 15 Jul 22:48 usr/x86_64-w64-mingw32.static/lib/libpoppler-glib.la
-rw-r--r--  1 tonyt  staff  1754398 15 Jul 22:48 usr/x86_64-w64-mingw32.static/lib/libpoppler-qt4.a
-rwxr-xr-x  1 tonyt  staff     3282 15 Jul 22:48 usr/x86_64-w64-mingw32.static/lib/libpoppler-qt4.la
-rw-r--r--  1 tonyt  staff  6938536 15 Jul 22:48 usr/x86_64-w64-mingw32.static/lib/libpoppler.a
-rwxr-xr-x  1 tonyt  staff     2957 15 Jul 22:48 usr/x86_64-w64-mingw32.static/lib/libpoppler.la
```

**Comment by darealshinji** [comment]
Rebased commits.

**Comment by tonytheodore** [comment]
Perfect, thanks!

**Pull request merged**